src/stream/src/executor/backfill/snapshot_backfill/executor.rs

```diff
@@ -449,19 +492,41 @@ impl<'a> UpstreamBuffer<'a, ConsumingSnapshot> {
     }
 
     fn start_consuming_log_store(
-        self,
-        snapshot_epoch: u64,
-    ) -> UpstreamBuffer<'a, ConsumingLogStore> {
+        mut self,
+        consumed_epoch: u64,
+    ) -> Option<UpstreamBuffer<'a, ConsumingLogStore>> {
+        // it's safe to access upstream_pending_barriers[0][0] because it's first upstream barrier
+        let first_upstream_barrier_epoch = self
+            .upstream_pending_barriers
+            .back()
+            .expect("first upstream barrier")
+            .back()
+            .expect("first upstream barrier")
+            .epoch;
```

Comment:
I think it's better to provide an interface for this type of access pattern. Might be a little hard to understand just looking at this.